account_fiscal_position_vies_warning/models/account_move.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2025 Moduon Team S.L. <info@moduon.team>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import ValidationError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def action_post(self):
+        if self.env.company.vat_check_vies:  # Verify VAT Numbers is True
+            company_id = self.env.company.id
+            try:
+                if (
+                    self.fiscal_position_id
+                    == self.env.ref(f"account.{company_id}_fp_intra")
+                    and not self.partner_id.vies_valid
+                ):
```

Comment:
Mmm... well, this is very dependant on `l10n_es` xml_ids, so it's very rigid IMO.

Maybe a more generic way would be to have a set of fiscal positions at company level and check them here